sol/main.c

```diff
@@ -90,63 +97,107 @@ insert_new_priority_req(struct req_queue *req_queue, struct priority_req *pr,
 		__func__);
 }
 
+/*
+ * Get the rte_mbuf struct for this entry.
+ * XXX This function should be part of DPDK.
+ */
+static inline struct rte_mbuf *
+rte_priv_to_mbuf(void *ptr) {
+	return RTE_PTR_SUB(ptr, sizeof(struct rte_mbuf));
+}
+
+static inline struct rte_mbuf *
+list_to_mbuf(struct list_head *list)
+{
+	return rte_priv_to_mbuf(list);
+}
+
+/*
+ * Get the first rte_mbuf element from a list.
+ * Note, that list is expected to be not empty.
+ */
+static inline struct rte_mbuf *
+list_first_entry_m(struct list_head *ptr) {
+	return rte_priv_to_mbuf(ptr->next);
```

Comment:
1. Code style: the opening braces of a function goes on its own line.
2. Use `list_to_mbuf()` here.
3. Items 1 and 2 also applies to `list_last_entry_m()`.